src/server/serve_dir.rs

```diff
@@ -22,43 +23,38 @@ impl ServeDir {
 impl<State> Endpoint<State> for ServeDir {
     fn call<'a>(&'a self, req: Request<State>) -> BoxFuture<'a, Result> {
         let path = req.uri().path();
-        let path = path.replacen(&self.prefix, "", 1);
+        let path = path.trim_start_matches(&self.prefix);
         let path = path.trim_start_matches('/');
-        let mut dir = self.dir.clone();
+        let mut file_path = self.dir.clone();
         for p in Path::new(path) {
-            dir.push(&p);
+            if p == OsStr::new(".") {
+                continue;
+            } else if p == OsStr::new("..") {
+                file_path.pop();
+            } else {
+                file_path.push(&p);
+            }
         }
-        log::info!("Requested file: {:?}", dir);
+
+        log::info!("Requested file: {:?}", file_path);
```

Comment:
Ideally we could `warn` here with the full requested path if it contained a `..` instead.